spring-boot-test-autoconfigure/src/test/java/org/springframework/boot/test/autoconfigure/jooq/JooqTestIntegrationTests.java

```diff
@@ -43,7 +41,6 @@
  */
 @RunWith(SpringRunner.class)
 @JooqTest
-@TestPropertySource(properties = "spring.jooq.sql-dialect = H2")
```

Comment:
I am confused. 

If the dialect is something mandatory (you have to provide it to be able to use Jooq), you can't just ignore that. You usually have a dialect in configuration with your database but you introduce a change with this PR that the database can be replaced by an embedded datasource.

So, if that is mandatory, it must have the right value without you having to specify that property.

Let's put that from a different perspective. If you support JOOQ in slicing, users are expecting that if the database gets replaced, things will follow along nicely (the dialect). If they customize the overriding so that the datasource is NOT replaced, same thing.

So IMO, that assertion must be there without the `@TestPropertySource` and we need another issue to auto-detect the dialect.